ui/src/utils/jupyterhub.test.ts

```diff
@@ -22,10 +22,16 @@ import {
   storeAppToStart,
 } from './jupyterhub';
 
-jest.mock('./jupyterhub', () => ({
-  ...jest.requireActual('./jupyterhub'),
-  navigateToUrl: jest.fn(),
-}));
+vi.mock('./jupyterhub', async () => {
+  // Require the actual module to spread its properties
+  const actual = await vi.importActual('./jupyterhub');
+
+  return {
+    ...actual,
+    navigateToUrl: vi.fn(),
+  };
+});
+
```

Comment:
Out of curiosity, do we have to make it async to make the test work? I thought Vitest is a drop-in replacement for Jest

Reply:
Looks like `vi.importActual` returns a promise, so it would have to either be async, or would need to chain with `then` to handle what happens when the promise is returned.

That said, in general, vite and jest provide a nearly identical implementation, but not 100%.

Reply:
I see, thanks for the explanation.